### PR TITLE
Add unit tests for runfolder utils

### PR DESCRIPTION
**Pull request by Aratz**
This PR implements a new class to manipulate and extract information from runfolders on disk.

### CODE REVIEW

**Review by Aratz — COMMENTED** [review]
Great work! My main comment is that `list_runfolders` should be able to ignore regular folders. Specifically, when a `Runfolder` instance is created on a regular folder, an `AssertionError` is raised. `list_runfolders` should ignore paths raising such an exception. I don't think it's possible to do this with list-comprehensions in a simple way, so you may need to use a regular for-loop instead.

Other comments:
- Unless you have strong opinions against it, I think you should keep `.gitattributes` and `tests/.pylintrc`:
  - `.gitattributes` tells git to avoid generating diffs for non human-readable files (or barely human-readable for .xml in this case)
  - `tests/.pylintrc` ignores specific linting rules for tests.

I will wait until this is merged before I review the other PRs, let me know if you want to pair program this or solve it on your own.

**Comment by nkongenelly** [comment]
> Great work! My main comment is that `list_runfolders` should be able to ignore regular folders. Specifically, when a `Runfolder` instance is created on a regular folder, an `AssertionError` is raised. `list_runfolders` should ignore paths raising such an exception. I don't think it's possible to do this with list-comprehensions in a simple way, so you may need to use a regular for-loop instead.


1. Should the list_runfolder ignore regular folders (raise an exception) and continue looking for Runfolders or should it stop whenever it finds a regular folder.

2. Then regarding raising the exceptions when it finds a regular folder, the current implementation will raise the Assertionerror on line 26 in the if condition (As the Runfolder instance is created and a regular folder will not have the Runparameters.xml file). Is this what was expected or is there something else that should be done to detect and ignore a regular folder?

**Comment by Aratz** [comment]
1. It should ignore regular folders, for example let's say the monitored directory contains the following folders:
```
monitored_directory
├── regular_folder
├── runfolder1
├── runfolder2
└── runfolder3
```
then `list_runfolders` should return `[Runfolder("runfolder1"), Runfolder("runfolder2"), Runfolder("runfolder3")]`, as specified in `test_list_runfolders`.

2.  `list_runfolders` should catch the exception that is raised when attempting to create a Runfolder from a regular folder and move on to the next path.

**Comment by nkongenelly** [comment]
@Aratz , thanks for that explanation. I have now pushed the changes.

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:24]:
```suggestion
                if filter_key(runfolder := Runfolder(os.path.join(monitored_directory, subdir))):
                    runfolders.append(runfolder)
```
You can use the [walrus operator](https://realpython.com/python-walrus-operator/) to avoid instantiating the same runfolder twice here :)

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:1]: I think it's better if we only use pathlib, i.e.:
- use `Path.iterdir` instead of `os.listdir`
- use the the `/` operator to combine paths instead of `os.path.join` (e.g. check https://github.com/arteria-project/arteria/blob/5e0ebe1a29b10c54d86fa25f64323cc4e7018cba/arteria/models/runfolder_utils.py#L56)

File: arteria/.gitattributes
Comment (Aratz) [inline arteria/.gitattributes:1]: This file should be placed on level above, at the root of the repository (instead of `arteria/`).

File: tests/unit/test_runfolder_utils.py
Comment (Aratz) [reply tests/unit/test_runfolder_utils.py]: I think it's a bit simpler if you restore this line rather than having a specific test for ignoring regular folders.

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:111]:
```suggestion
```

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:149]: Update docstring here.

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:142]:
```suggestion
        completed_marker_file = instrument_keys["completed_marker_file"]
```
This function should always return a valid marker file (and never `None`).

File: arteria/models/runfolder_utils.py
Comment (Aratz) [inline arteria/models/runfolder_utils.py:176]:
```suggestion
                raise TypeError(f"{self.instrument} is not recognized.") from e
```

**Review by Aratz — COMMENTED** [review]
Thanks for addressing my comments! I have a few more minor suggestions and then I think we'll be good to go :)

**Comment by nkongenelly** [comment]
Thanks for the suggested changes. I've learnt a lot from the reviews so far.

I have now updated the code.

**Comment by Aratz** [comment]
Great! I think it's good to go now :) Unfortunately I can't click "approve" because I created this PR. If you want you can approve it yourself and merge it since you're set as reviewer

**Review by nkongenelly — APPROVED** [review]

**Pull request merged**
